test/CollectibleV1.t.sol

```diff
@@ -116,3 +116,50 @@ contract RemoteBurnTest is CollectibleV1Test {
         assertEq(collectibleV1.totalSupply(), maxSupply - 1);
     }
 }
+
+contract SafeBatchTransferFromTest is CollectibleV1Test {
+    function setUp() public virtual override {
+        CollectibleV1Test.setUp();
+    }
+
+    function test_RevertWhen_NotAuthorized() public {
+        uint256[] memory ids = new uint256[](1);
+        ids[0] = 0;
+        vm.expectRevert(BaseToken.BaseToken_NotAuthorized.selector);
+        collectibleV1.safeBatchTransferFrom(accounts[0], accounts[1], ids, "");
+    }
+
+    function test_RevertWhen_ReceiverAddressIsZero() public {
+        // ensure sender owns a token
+        vm.prank(deployer);
+        collectibleV1.mintTo(accounts);
+
+        uint256[] memory ids = new uint256[](1);
+        ids[0] = 0;
+
+        vm.prank(accounts[0]);
+        vm.expectRevert(bytes("ERC721: transfer to the zero address"));
+        collectibleV1.safeBatchTransferFrom(accounts[0], address(0), ids, "");
+    }
+
+    function test_SafeBatchTransferFrom() public {
+        // ensure sender owns a token
+        vm.prank(deployer);
+        address[] memory sameAddresses = new address[](3);
+        sameAddresses[0] = accounts[0];
+        sameAddresses[1] = accounts[0];
+        sameAddresses[2] = accounts[0];
+        collectibleV1.mintTo(sameAddresses);
+
+        uint256[] memory ids = new uint256[](3);
+        ids[0] = 0;
+        ids[1] = 1;
+        ids[2] = 2;
+
+        vm.prank(accounts[0]);
+        collectibleV1.safeBatchTransferFrom(accounts[0], accounts[1], ids, "");
+
+        assertEq(collectibleV1.balanceOf(accounts[0]), 0);
+        assertEq(collectibleV1.balanceOf(accounts[1]), 3);
+    }
```

Comment:
Can probably add some more tests here.

Reply:
I've updated the `RevertWhen_NotAuthorized` test to check for the approvals case.